proc/proc.go

```diff
@@ -149,24 +149,25 @@ func GetKernelModules(modulesPath string,
 }
 
 // ListPIDs from the proc filesystem mount point and return a list of util.PID to be processed
-func ListPIDs() ([]libpf.PID, error) {
-	pids := make([]libpf.PID, 0)
-	files, err := os.ReadDir(defaultMountPoint)
+func ListPIDs(callback func(pid libpf.PID) error) error {
+	dir, err := os.Open(defaultMountPoint)
 	if err != nil {
-		return nil, err
+		return err
 	}
-	for _, f := range files {
-		// Make sure this is a PID file entry
-		if !f.IsDir() {
-			continue
-		}
-		pid, err := strconv.ParseUint(f.Name(), 10, 32)
+	filenames, err := dir.Readdirnames(-1)
```

Comment:
`Readdirnames` may fail and return partial results in `filenames` with an error, in which case we should probably continue processing `filenames` (but also return the error).

Reply:
Good point.
Though, if we return an error here, the agent currently stops. This would not be OK for an intermediate error. So in case of a partial result, let's log the error message as warning and continue. WDYT?

Reply:
The agent shouldn't [stop](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/c088a0582883ab1d04f28b562de32c4006f55b83/main.go#L226) if this errors.

Either way, logging the error is fine as we're calling this function for side-effects. If it was still returning a slice of PIDs, we could let the caller decide what to do and return both an error and the partial slice of results.

Reply:
Added a commit that returns a bool to indicate a partial result. Now the decision of logging / stopping is made by the caller. This also removes the hard-coded `/proc` string from the caller (which actually isn't known at that point).